File: php/Shirt.php
<?php

require_once 'Clothing.php'; 

class Shirt extends Clothing {

    private $color;
    private $sleeve_type;

    public function __construct($id = "", $name = "", $brand = "", $price = 0, $size = "", $material = "", $gender = "", $color = "", $sleeve_type = "") {
        parent::__construct($id, $name, $brand, $price, $size, $material, $gender);
        $this->color = $color;
        $this->sleeve_type = $sleeve_type;
    }

    // Getter and setter methods for color
    public function get_color() {
        return $this->color;
    }

    public function set_color($color) {
        $this->color = $color;
    }

    // Getter and setter methods for sleeve_type
    public function get_sleeve_type() {
        return $this->sleeve_type;
    }

    public function set_sleeve_type($sleeve_type) {
        $this->sleeve_type = $sleeve_type;
    }
}
  

?>

File: cpp/program/Shirt.cpp
#include <iostream>
#include <string>
#include "Clothing.cpp"

class Shirt : public Clothing {
private:
    std::string color;
    std::string sleeveType;

public:
    // Constructor
    Shirt(const std::string& id = "", const std::string& name = "", const std::string& brand = "",
          double price = 0.0, const std::string& size = "", const std::string& material = "",
          const std::string& gender = "", const std::string& color = "", const std::string& sleeveType = "")
        : Clothing(id, name, brand, price, size, material, gender), color(color), sleeveType(sleeveType) {}

    // Getter methods
    std::string getColor() const { return color; }
    std::string getSleeveType() const { return sleeveType; }

    // Setter methods
    void setColor(const std::string& newColor) { color = newColor; }
    void setSleeveType(const std::string& newSleeveType) { sleeveType = newSleeveType; }
};
